File: app/src/main/java/com/example/lab6dev/pizza/dao/PizzaDao.java
package com.example.lab6dev.pizza.dao;

import com.example.lab6dev.pizza.model.PizzaRecipe;
import java.util.List;

// Fait par Abdelghafour
public interface PizzaDao {
    PizzaRecipe addRecipe(PizzaRecipe recipe);
    PizzaRecipe updateRecipe(PizzaRecipe recipe);
    boolean deleteRecipe(long id);
    PizzaRecipe getRecipeById(long id);
    List<PizzaRecipe> getAllRecipes();
}

File: app/src/main/java/com/example/lab6dev/pizza/dao/PizzaDao.kt
package com.example.lab6dev.pizza.dao

import com.example.lab6dev.pizza.model.PizzaRecipe

// Fait par Abdelghafour
interface PizzaDao {
    fun addRecipe(recipe: PizzaRecipe): PizzaRecipe
    fun updateRecipe(recipe: PizzaRecipe): PizzaRecipe?
    fun deleteRecipe(id: Long): Boolean
    fun getRecipeById(id: Long): PizzaRecipe?
    fun getAllRecipes(): List<PizzaRecipe>
}
